### PR TITLE
docker-gc: don't run `system prune`

### PR DESCRIPTION
**Pull request by jkoelker**
## High-level description

`docker system prune` will remove networks that do not have containers
currently attached to them. This will remove the `dcos` and `dcos6`
network if they are not in use. Instead prune just the objects desired.

## Corresponding DC/OS tickets (required)

  - [DCOS_OSS-5441](https://jira.mesosphere.com/browse/DCOS_OSS-5441) Switch to `docker system prune`.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by jkoelker** [comment]
@mesosphere-mergebot changelog-not-required service ordering fix for regression from #6454

**Comment by jkoelker** [comment]
@mesosphere-mergebot request-review @drewvanstone @br-lewis @makkes @bamarni @rgo3 @janisz @urbanserj

File: packages/docker-gc/extra/dcos-docker-gc.service [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Docker GC: periodically garbage collects Docker containers and images
 ConditionPathExists=/opt/mesosphere/etc/docker_gc_enabled
+Before=dcos-mesos-slave.service
```
Comment (urbanserj) [inline packages/docker-gc/extra/dcos-docker-gc.service]: https://github.com/dcos/dcos/blob/master/docs/systemd-rules.md

Comment (urbanserj) [reply packages/docker-gc/extra/dcos-docker-gc.service]: > `Before=`, `After=` are discouraged. They are not strong guarantees

Comment (janisz) [reply packages/docker-gc/extra/dcos-docker-gc.service]: Can we use filter to not remove dcos networks?

Comment (jkoelker) [reply packages/docker-gc/extra/dcos-docker-gc.service]: :+1: while I agree in general, I would advocate that this be one of the exceptions, as it stands its a coin flip as to if docker apps can use container networks.

Comment (urbanserj) [reply packages/docker-gc/extra/dcos-docker-gc.service]: @janisz I'd probably run `docker container prune`, `docker image prune`, etc, but not `docker network prune` or `docker system prune`.

Comment (jkoelker) [reply packages/docker-gc/extra/dcos-docker-gc.service]: > Can we use filter to not remove dcos networks?

Not currently, as no labels are added to the network to be able to filter on

Comment (jkoelker) [reply packages/docker-gc/extra/dcos-docker-gc.service]: @urbanserj @janisz I like the idea of just running the `container` and `image` `prunes`. If that's amenable to all, I'll change to that here.

**Comment by urbanserj** [comment]
@mesosphere-mergebot changelog-not-required bug fix

**Pull request merged**